reference/reference.go

```diff
@@ -168,6 +169,20 @@ func ParseIDOrReference(idOrRef string) (digest.Digest, Named, error) {
 	return "", ref, err
 }
 
+// Match reports whether ref matches the specified pattern
+func Match(pattern string, ref Named) bool {
+	// FIXME: should we support pattern for NamedTagged or not ?
```

Comment:
First, we should implement this in the upstream distribution package. These packages keep diverging and it is extremely disappointing.

As far as support goes, we should support globbing over any of the reference types.